src/query/storages/hive/src/hive_blocks.rs

```diff
@@ -0,0 +1,73 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use common_arrow::parquet::metadata::FileMetaData;
+use common_arrow::parquet::metadata::RowGroupMetaData;
+
+use crate::HivePartInfo;
+
+#[derive(Clone, Debug)]
+pub struct HiveBlocks {
+    pub file_meta: FileMetaData,
+    pub part: HivePartInfo,
+    pub valid_rowgroups: Vec<usize>,
+    pub current_index: usize,
+}
+
+impl HiveBlocks {
+    pub fn create(file_meta: FileMetaData, part: HivePartInfo) -> Self {
+        Self {
+            file_meta,
+            part,
+            valid_rowgroups: vec![],
+            current_index: 0,
+        }
+    }
+
+    // there are some conditions to filter invalid row_groups:
+    // 1. the rowgroup doesn't belong to the partition
+    // 2. filtered by predict pushdown (todo)
+    pub fn prune(&mut self) -> bool {
```

Comment:
Can we read the `file_meta` in `read_partition` method?

Thus we can split the row-groups there without introducing `hive_file_splitter`.

Reply:
it'll be too slow to read file meta in read_partition if there're too many files.

Reply:
So we can parallel this in `read_partition` and cache the meta in part info?

Now seems we may read meta per file more than once.